### PR TITLE
Allow JUnitLauncher to launch all tests in a Package including subpackages

### PR DESCRIPTION
**Pull request by RoiSoleil**
## What it does
Allow JUnitLauncher to launch all tests in a Package including subpackages.

Fix : https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/420

## How to test
Create a package "a" without test and add a "a.b" with test "C" and in Eclipse run test on "a".
Verify that "C" is run.

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by RoiSoleil** [comment]
> /rebase

Done

**Review by jjohnstn — COMMENTED** [review]
Works well.  I just had one minor nit comment and a question about the override of the StandardJavaElementContentProvider.

File: org.eclipse.jdt.junit/src/org/eclipse/jdt/junit/launcher/JUnitLaunchConfigurationTab.java [hunk 1]
```diff
@@ -1153,20 +1153,21 @@ public boolean isSelectedValid(Object element) {
...
-				}
 				return super.select(viewer, parent, element);
 			}
 		};
 
-		StandardJavaElementContentProvider provider= new StandardJavaElementContentProvider();
+		StandardJavaElementContentProvider provider= new StandardJavaElementContentProvider() {
+			@Override
+			public boolean hasChildren(Object element) {
+				if(element instanceof IPackageFragment) {
```
Comment (jjohnstn) [inline org.eclipse.jdt.junit/src/org/eclipse/jdt/junit/launcher/JUnitLaunchConfigurationTab.java:1165]: What does this do?  I tried without it and it works the same.

Comment (RoiSoleil) [reply org.eclipse.jdt.junit/src/org/eclipse/jdt/junit/launcher/JUnitLaunchConfigurationTab.java:1165]: It fix a bug in UI. The packages show a button to expand but the contentprovider will never provide children for a package in this context.

image

Comment (jjohnstn) [reply org.eclipse.jdt.junit/src/org/eclipse/jdt/junit/launcher/JUnitLaunchConfigurationTab.java:1165]: Ah, in the Configuration dialog.  Ok.

**Comment by jjohnstn** [comment]
Thanks @RoiSoleil

**Pull request merged**